lib/dialects/sqlite/connection-manager.js

```diff
@@ -44,19 +46,27 @@ class ConnectionManager extends AbstractConnectionManager {
   getConnection(options) {
     options = options || {};
     options.uuid = options.uuid || 'default';
-    options.inMemory = (this.sequelize.options.storage || this.sequelize.options.host || ':memory:') === ':memory:' ? 1 : 0;
+    options.storage = this.sequelize.options.storage || this.sequelize.options.host || ':memory:';
+    options.inMemory = options.storage === ':memory:' ? 1 : 0;
 
     const dialectOptions = this.sequelize.options.dialectOptions;
-    options.readWriteMode = dialectOptions && dialectOptions.mode;
+    options.readWriteMode = dialectOptions && dialectOptions.mode || this.lib.OPEN_READWRITE | this.lib.OPEN_CREATE; // default mode;
```

Comment:
Can you please refactor this for readability?

```suggestion
    const defaultReadWriteMode = this.lib.OPEN_READWRITE | this.lib.OPEN_CREATE;
    options.readWriteMode = dialectOptions && dialectOptions.mode || defaultReadWriteMode;
```